gcloud/datastore/batch.py

```diff
@@ -159,7 +163,11 @@ def commit(self):
         however it can be called explicitly if you don't want to use a
         context manager.
         """
-        self.connection.commit(self._dataset_id, self.mutation)
+        response = self.connection.commit(self._dataset_id, self.mutation)
+        for new_key_pb, entity in zip(response.insert_auto_id_key,
+                                      self._auto_id_entities):
```

Comment:
Yes the order will be the same. The commit will either all fail or all succeed, so no partial success.